tests/ctst/common/common.ts

```diff
@@ -300,40 +300,68 @@ Then('i {string} be able to add user metadata to object {string}',
-                `Topic ${topics[i]} partition ${j} offset has not increased,
-                previousOffsets: ${previousOffsets[i].partitions[j].low} / ${previousOffsets[i].partitions[j].high},
-                newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
+
+
+        while (topics.length > 0) {
+            const previousOffsets = await getTopicsOffsets(topics, kafkaAdmin);
+            // Checking topics offsets before kafkacleaner passes to be sure kafkacleaner works
+            // This function can be improved by consuming messages and
+            // verify that the timestamp is not older than last kafkacleaner run
+            // Instead of waiting for a fixed amount of time,
+            // we could also check for metrics to see last kafkacleaner run
+
+            // 3 seconds added to be sure kafkacleaner had time to process
+            await Utils.sleep(kfkcIntervalSeconds * 1000 + 3000);
+
+            const newOffsets = await getTopicsOffsets(topics, kafkaAdmin);
+
+            for (let i = 0; i < topics.length; i++) {
+                process.stdout.write(`\nChecking topic ${topics[i]}\n`);
+                for (let j = 0; j < newOffsets[i].partitions.length; j++) {
+                    const newMessagesAfterClean =
+                        newOffsets[i].partitions[j].low === previousOffsets[i].partitions[j].high &&
+                        previousOffsets[i].partitions[j].high !== '0';
+
+                    if (newMessagesAfterClean) {
+                        // If new messages appeared after we gathered the offsets, we need to recheck after
+                        process.stdout.write(`New messages after clean for topic ${topics[i]} rechecking after`);
+                        continue;
+                    }
+
+                    const lowOffsetIncreased = newOffsets[i].partitions[j].low > previousOffsets[i].partitions[j].low;
+                    const allMessagesCleaned = newOffsets[i].partitions[j].high === newOffsets[i].partitions[j].low;
+
+                    // If the low offset increased it means the topic has been cleaned
+                    // If low offset is the same as high offset,
+                    // it means the topic is completly cleaned even though lowOffset didnt increased
+                    assert.ok(lowOffsetIncreased || allMessagesCleaned,
+                        `Topic ${topics[i]} partition ${j} offset has not increased,
+                        previousOffsets: ${previousOffsets[i].partitions[j].low} /\
+                         ${previousOffsets[i].partitions[j].high},
+                        newOffsets: ${newOffsets[i].partitions[j].low} / ${newOffsets[i].partitions[j].high}`);
+
+                    // Topic is cleaned, we don't need to check it anymore
+                    topics.splice(i, 1);
+                }
             }
         }
+        clearTimeout(timeoutID);
```

Comment:
in case of `assert` before, this will not be executed : which could cause strange behavior.

a safer approach would be to use `try{}finally{}` block:

```suggestion
        const timeoutID = setTimeout(....);
        try {
           //run the check
        } finally {
           clearTimeout(timeoutID);
        }
```